client/src/context/auth/AuthState.js

```diff
@@ -0,0 +1,110 @@
+import React, { useReducer } from 'react';
+import axios from 'axios';
+
+import AuthContext from './authContext';
+import authReducer from './authReducer';
+
+import setAuthToken from '../../utils/setAuthToken';
+
+import {
+  REGISTER_SUCCESS,
+  REGISTER_FAIL,
+  USER_LOADED,
+  AUTH_ERROR,
+  LOGIN_SUCCESS,
+  LOGIN_FAIL,
+  LOGOUT,
+  CLEAR_ERRORS,
+} from '../types.js';
+
+const AuthState = (props) => {
+  const initialState = {
+    token: localStorage.getItem('token'),
+    isAuthenticated: null,
+    loading: true,
+    error: null,
+    user: null,
+  };
+  const [state, dispatch] = useReducer(authReducer, initialState);
+
+  // Load User
+  const loadUser = async () => {
+    if (localStorage.token) {
+      setAuthToken(localStorage.token);
+    }
+    try {
+      const res = await axios.get('/api/auth');
+      dispatch({ type: USER_LOADED, payload: res.data });
+    } catch (error) {
+      dispatch({ type: AUTH_ERROR });
+    }
+  };
+  // Register User
+  const register = async (formData) => {
+    const config = {
+      headers: { 'Content-Type': 'application/json' },
+    };
+    try {
+      const res = await axios.post('/api/users', formData, config);
+      dispatch({
+        type: REGISTER_SUCCESS,
+        payload: res.data,
+      });
+      loadUser();
+    } catch (err) {
+      dispatch({
+        type: REGISTER_FAIL,
+        payload: err.response.data.msg,
```

Comment:
what if err doesn't have response...response doesn't have data and data doesn't have mgs? hahah
Validations on dot operators please :) , else your catch will throw errors